File: server/src/main/java/de/hsrm/mi/swtpro03/server/network/messages/game/GameEndMessage.java
package de.hsrm.mi.swtpro03.server.network.messages.game;


import java.util.List;
import java.util.Map;

public class GameEndMessage {

    public static final String TYPE = "GameEndMessage";

    public static final String WIN_STATUS = "WIN";
    public static final String LOSS_STATUS = "LOSS";
    public static final String QUIT_STATUS = "QUIT";

    private Map<String, Integer> scores;
    private String status;
    private String winnerColor;

    public GameEndMessage() {
    }

    public GameEndMessage(Map<String, Integer> scores, String status, String winnerColor) {
        this.scores = scores;
        this.status = status;
        this.winnerColor = winnerColor;
    }

    public Map<String, Integer> getScores() {
        return scores;
    }

    public void setScores(Map<String, Integer> scores) {
        this.scores = scores;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getWinnerColor() {
        return winnerColor;
    }

    public void setWinnerColor(String winnerColor) {
        this.winnerColor = winnerColor;
    }
}

File: client/Client/Network/Messages/Game/GameEndMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LoadRunnerClient.Network.Messages
{
    class GameEndMessage
    {
        public const string TYPE = "GameEndMessage";
        public const string WIN_STATUS = "WIN";
        public const string LOSS_STATUS = "LOSS";
		public const string QUIT_STATUS = "QUIT";
		private Dictionary<string, int> _Scores;
		private string winnerColor;
		public Dictionary<string, int> Scores
        {
            get => _Scores;
            set => _Scores = value;
        }

        private string _status;
        public string Status
        {
            get => _status;
            set => _status = value;
        }
		public string WinnerColor { get => winnerColor; set => winnerColor = value; }

		public GameEndMessage() { }

    }
}
